src/h_assets/cached_file.py

```diff
@@ -0,0 +1,75 @@
+import configparser
+import json
+from os.path import getmtime
+
+from pyramid.settings import aslist
+
+
+class CachedFile:  # pylint:disable=too-few-public-methods
+    """
+    Parses content from a file and caches the result.
+
+    _CachedFile reads a file at a given path and parses the content using a
+    provided loader.
+    """
+
+    path = None
+
+    def __init__(self, path, auto_reload):
+        """
+        Create the CachedFile object.
+
+        :param path: The path to the file to load
+        :param loader: A callable that parses content from a file
+        :param auto_reload: If True, the parsed content is discarded if the
+            mtime of the file changes.
+        """
+
+        self.path = path
+        self._auto_reload = auto_reload
+        self._last_modified_time = None
+        self._cached_content = None
+
+    def load(self):
+        """
+        Return the content of the file parsed with the loader.
```

Comment:
This comment is out of date, since the constructor no longer accepts a loader function.

Reply:
I've given all of the doc strings a once over